src/OpenTracing/SpanReference.php

```diff
@@ -2,20 +2,93 @@
 
 namespace OpenTracing;
 
-interface SpanReference
+use OpenTracing\Exceptions\InvalidReferenceArgument;
+
+final class SpanReference
 {
+    const CHILD_OF = 'child_of';
+    const FOLLOWS_FROM = 'follows_from';
+
     /**
-     * @return bool
+     * @var SpanContext
      */
-    public function isTypeChildOf();
+    private $context;
 
     /**
-     * @return bool
+     * @var string
+     */
+    private $type;
+
+    private function __construct(SpanContext $context, $type)
+    {
+        $this->context = $context;
+        $this->type = $type;
+    }
+
+    /**
+     * @param SpanContext|Span $context
+     * @param string $type
+     * @throws InvalidReferenceArgument on empty type
+     * @return SpanReference when context is invalid
+     */
+    public static function create($context, $type)
+    {
+        if (empty($type)) {
+            throw InvalidReferenceArgument::emptyType();
```

Comment:
Since the constructor is private, I'd rather add this check in the public API rather than adding for those cases where I know that I am actually passing a valid string (e.g `createAsChildOf` or `createAsFollowsFrom`), that is why I don't add this check in the constructor.